config/config.go

```diff
@@ -17,117 +17,98 @@ package config
-
-// defaultTabletType is a default Vitess tablet type.
-var defaultTabletType = strings.ToLower(topodata.TabletType_name[int32(topodata.TabletType_PRIMARY)])
-
 const (
 	// KeyAddress is a config name for an address.
 	KeyAddress = "address"
-	// KeyTable is a config name for an table.
-	KeyTable = "table"
 	// KeyUsername is a config name for an username.
 	KeyUsername = "username"
 	// KeyPassword is a config name for an password.
 	KeyPassword = "password"
-	// KeyKeyspace is a config name for a keyspace.
-	KeyKeyspace = "keyspace"
-	// KeyTabletType is a config name for a tabletType.
-	KeyTabletType = "tabletType"
-	// KeyMaxRetries is a config name for a grpc max retries.
-	KeyMaxRetries = "maxRetries"
-	// KeyRetryTimeout is a config name for grpc retry timeout.
-	KeyRetryTimeout = "retryTimeout"
 )
 
-const (
-	// DefaultRetryTimeout is a default timeout that is used
-	// when a timeout provided to the DialWithRetries function is less or equal to zero.
-	DefaultRetryTimeout = time.Second
+// ErrUnknownTabletType occurs when a provided tablet type is not valid.
+var ErrUnknownTabletType = errors.New("unknown tablet type")
 
-	// DefaultMaxRetries is a default retries, that given to dial to vitess grpc server.
-	DefaultMaxRetries = 3
-)
+// defaultTabletType is a default Vitess tablet type.
+var defaultTabletType = strings.ToLower(topodata.TabletType_name[int32(topodata.TabletType_PRIMARY)])
 
 // Config contains configurable values
 // shared between source and destination Vitess connector.
 type Config struct {
 	// Address is an address pointed to a VTGate instance.
-	Address string `key:"address" validate:"required,hostname_port"`
+	Address string `json:"address" validate:"required"`
 	// Table is a name of the table that the connector should write to or read from.
 	// Max length is 64, see Identifier Length Limits
 	// https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html
-	Table string `key:"table" validate:"required,max=64"`
+	Table string `json:"table" validate:"required"`
 	// Keyspace specifies a VTGate keyspace.
-	Keyspace string `key:"keyspace" validate:"required"`
+	Keyspace string `json:"keyspace" validate:"required"`
 	// Username is a username of a VTGate user.
-	Username string `key:"username" validate:"required_with=Password"`
+	Username string `json:"username"`
 	// Password is a password of a VTGate user.
-	Password string `key:"password" validate:"required_with=Username"`
+	Password string `json:"password"`
 	// TabletType is a tabletType.
-	TabletType string `key:"tabletType"`
+	TabletType string `json:"tabletType"`
 	// MaxRetries is the number of reconnect retries the connector will make before giving up if a connection goes down.
-	MaxRetries int `key:"maxRetries"`
+	MaxRetries int `json:"maxRetries" default:"3"`
 	// RetryTimeout is the time period that will be waited between retries.
-	RetryTimeout time.Duration `key:"retryTimeout" validate:"gte=0"`
```

Comment:
Nitpick: we have a gt validation in the SDK.

Reply:
Yes, we have default value as 1s so that should handle that.

Reply:
I meant, what if a user enters "-5m"?

Reply:
fixed.